hub/api/v1/neardata.py

```diff
@@ -0,0 +1,224 @@
+                            "receipt_id": receipt_execution_outcome["execution_outcome"]["id"],
+                        },
+                        auth_token,
+                    )
+
+
+def load_auth_token():
+    from nearai.config import Config, load_config_file
+
+    app_config = Config()
+    # Update config from global config file
+    config_data = load_config_file(local=False)
+    app_config = app_config.update_with(config_data)
+
+    return app_config.auth
+
+
+async def run_agent(agent, message, signer_id, data, auth_token: AuthToken):
+    if not (agent and message):
+        return print("Missing data")
+
+    # TODO find out how to ret fid of 401 error here
+    # auth_token.on_behalf_of = signer_id
+
+    thread_model = ThreadModel(
+        meta_data={
+            "agent_ids": [agent],
+        },
+        tool_resources=None,
+        owner_id=auth_token.account_id,
+    )
+
+    thread = _create_thread(thread_model, auth=auth_token)
+
+    run_params = RunCreateParamsBase(
+        assistant_id=agent,
+        model=DEFAULT_MODEL,
+        instructions=None,
+        tools=None,
+        metadata=None,
+        include=[],
+        additional_instructions=None,
+        additional_messages=[{"content": message, "role": "user"}],
+        max_completion_tokens=None,
+        max_prompt_tokens=None,
+        parallel_tool_calls=None,
+        response_format=None,
+        temperature=None,
+        tool_choice=None,
+        top_p=None,
+        truncation_strategy=None,
+        stream=False,
+        schedule_at=None,
+        delegate_execution=False,
+        parent_run_id=None,
+        # max_iterations=data.get("max_iterations", 1),
+    )
+
+    create_run(thread_id=thread.id, run=run_params, auth=auth_token, scheduler=scheduler)
+
+
+# Main asynchronous function to process blocks
+async def process_blocks(auth_token):
+    # Get the last processed blockId
+    block_id = await get_last_block_id()
+
+    # Try to load the next block
+    new_block_id = await load_block(block_id + 1, auth_token)
+    if new_block_id:
+        # If block is found, save the new blockId
+        print(f"New block found: {new_block_id}")
+        save_last_block_id(new_block_id)
+        return True
+    else:
+        print(f"Block not found: {block_id}")
+        return False
+
+
+async def periodic_task(auth_token: AuthToken):
+    # Try to catch up reading NUMBER_OF_BLOCKS_TO_READ_IF_NOT_SYNCED blocks per run
+    for _ in range(NUMBER_OF_BLOCKS_TO_READ_IF_NOT_SYNCED):
+        block_found = await process_blocks(auth_token)
+        if not block_found:
+            break
+
+
+@asynccontextmanager
+async def lifespan(app):
+    # remove last known block_id to reset the state on start
+    if RESET_BLOCK_ID_ON_START:
+        if os.path.exists(BLOCK_ID_FILE):
+            os.remove(BLOCK_ID_FILE)
+            print(f"File {BLOCK_ID_FILE} has been deleted.")
+
+    auth_token = load_auth_token()
```

Comment:
Is this what you are planning to change to a 'Hub auth token'? Even though it would be convenient to load it from config I think we should load if from somewhere else since this is a common area of confusion. It would be easy for a developer to write functionality that uses the hub auth when what they need is the user's auth.

Reply:
Yes, that's a good point. Yes, this is the Hub auth token. If we separate it from the general config, are you suggesting creating a separate file and instructions on how to set up a dedicated config? I feel that might be more complicated than adding a section to the instructions on how to run the local hub, where we can clarify the purpose of the token in this case.

Reply:
Maybe the Hub token can go in `hub/.env` since it is the hub's auth and therefore be distinct from user auth in the config.json? The .env.example can refer to it, the 'how to run a local hub' docs can explain how to create one. We can then add a `sign_as_hub` function to neardata.py or x_event_source that pulls it from the environment and uses it..

Reply:
Please take a look, I just added docs to clarify it for agents builders and hub maintainers.